x/group/genesis_test.go

```diff
@@ -0,0 +1,507 @@
+		},
+		SubmittedAt: submittedAt,
+		Status:      ProposalStatusClosed,
+		Result:      ProposalResultAccepted,
+		VoteState: Tally{
+			YesCount:     "1",
+			NoCount:      "0",
+			AbstainCount: "0",
+			VetoCount:    "0",
+		},
+		Timeout:        timeout,
+		ExecutorResult: ProposalExecutorResultSuccess,
+	}
+	err = proposal.SetMsgs([]sdk.Msg{&banktypes.MsgSend{
+		FromAddress: accAddr.String(),
+		ToAddress:   memberAddr.String(),
+		Amount:      sdk.Coins{sdk.NewInt64Coin("test", 100)},
+	}})
+	require.NoError(t, err)
+
+	testCases := []struct {
+		name         string
+		genesisState GenesisState
+		expErr       bool
+	}{
+		{
+			"valid genesisState",
+			GenesisState{
+				GroupSeq:              2,
+				Groups:                []*GroupInfo{{GroupId: 1, Admin: accAddr.String(), Metadata: []byte("1"), Version: 1, TotalWeight: "1"}, {GroupId: 2, Admin: accAddr.String(), Metadata: []byte("2"), Version: 2, TotalWeight: "2"}},
+				GroupMembers:          []*GroupMember{{GroupId: 1, Member: &Member{Address: memberAddr.String(), Weight: "1", Metadata: []byte("member metadata")}}, {GroupId: 2, Member: &Member{Address: memberAddr.String(), Weight: "2", Metadata: []byte("member metadata")}}},
+				GroupPolicyAccountSeq: 1,
+				GroupPolicies:         []*GroupPolicyInfo{groupPolicy},
+				ProposalSeq:           1,
+				Proposals:             []*Proposal{proposal},
+				Votes:                 []*Vote{{ProposalId: proposal.ProposalId, Voter: memberAddr.String(), SubmittedAt: submittedAt, Choice: Choice_CHOICE_YES}},
+			},
+			false,
+		},
+		{
+			"empty genesisState",
+			GenesisState{},
+			false,
+		},
+		{
+			"empty group id",
+			GenesisState{
+				Groups: []*GroupInfo{
+					{
+						GroupId:     0,
+						Admin:       accAddr.String(),
+						Metadata:    []byte("1"),
+						Version:     1,
+						TotalWeight: "1",
+					},
+				},
+			},
+			true,
+		},
+		{
+			"invalid group admin",
+			GenesisState{
+				Groups: []*GroupInfo{
+					{
+						GroupId:     1,
+						Admin:       "invalid admin",
+						Metadata:    []byte("1"),
+						Version:     1,
+						TotalWeight: "1",
+					},
+				},
+			},
+			true,
+		},
+		{
+			"invalid group version",
+			GenesisState{
+				Groups: []*GroupInfo{
+					{
+						GroupId:     1,
+						Admin:       accAddr.String(),
+						Metadata:    []byte("1"),
+						Version:     0,
+						TotalWeight: "1",
+					},
+				},
+			},
+			true,
+		},
+		{
+			"invalid group TotalWeight",
+			GenesisState{
+				Groups: []*GroupInfo{
+					{
+						GroupId:     1,
+						Admin:       accAddr.String(),
+						Metadata:    []byte("1"),
+						Version:     1,
+						TotalWeight: "-1",
+					},
+				},
+			},
+			true,
+		},
+		{
+			"invalid group policy address",
+			GenesisState{
+				GroupPolicies: []*GroupPolicyInfo{
+					{
+						Address:  "invalid address",
+						GroupId:  1,
+						Admin:    accAddr.String(),
+						Version:  1,
+						Metadata: []byte("policy metadata"),
+					},
+				},
+			},
+			true,
+		},
```

Comment:
this is actually erroring on `if _, exists := groups[g.GroupId]; !exists {` and not on `GroupPolicyInfo.ValidateBasic`
you need to add Group with id 1 to the genesis state to test it properly
Same for all the test cases below